src/include/connection.h

```diff
@@ -287,6 +287,29 @@ typedef TAILQ_HEAD(__wt_backuphash, __wt_backup_target) WT_BACKUPHASH;
 
 extern const WT_NAME_FLAG __wt_stress_types[];
 
+#define WT_SESSION_FOREACH_BEGIN(array_session, conn)                                            \
+    do {                                                                                         \
+        uint32_t __i, __session_cnt;                                                             \
+        u_int __active;                                                                          \
+                                                                                                 \
+        /*                                                                                       \
+         * The session count variable can change concurrently with the reader thread traversing  \
+         * the array. It is up to the algorithm doing the traversal, e.g. hazard pointers,       \
+         * generations, etc, to handle the consequences of a new session being created or closed \
+         * out.                                                                                  \
+         */                                                                                      \
+        __session_cnt = *(volatile uint32_t *)&((conn)->session_cnt);                            \
+        for (__i = 0, (array_session) = (conn)->sessions; __i < __session_cnt;                   \
+             ++__i, ++(array_session)) {                                                         \
+            WT_ORDERED_READ(__active, (array_session)->active);                                  \
```

Comment:
It isn't clear to me why `active` needs to be an ordered read. I don't think it matters if multiple iterations of this loop happen out of order for different session handles, which is what I think the ordered read would enforce.

Is the concern that other fields from the session handle in this iteration of the loop (e.g of `txn` would be done out of order compared to this read?)

Reply:
I'll add a comment within the macro to describe this but yes you are correct that we are ordering against other fields in the session. Which is your second point. Basically writer side:
```
init session
write barrier
active = 1
```
Therefore reader side:
```
read active
read barrier
read session fields.
```
It's a fairly standard barrier use case.